genai-perf/genai_perf/export_data/prometheus_exporter.py

```diff
@@ -0,0 +1,83 @@
+# Copyright 2025, NVIDIA CORPORATION & AFFILIATES. All rights reserved.
+#
+# Redistribution and use in source and binary forms, with or without
+# modification, are permitted provided that the following conditions
+# are met:
+#  * Redistributions of source code must retain the above copyright
+#    notice, this list of conditions and the following disclaimer.
+#  * Redistributions in binary form must reproduce the above copyright
+#    notice, this list of conditions and the following disclaimer in the
+#    documentation and/or other materials provided with the distribution.
+#  * Neither the name of NVIDIA CORPORATION nor the names of its
+#    contributors may be used to endorse or promote products derived
+#    from this software without specific prior written permission.
+#
+# THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS ``AS IS'' AND ANY
+# EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+# IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR
+# PURPOSE ARE DISCLAIMED.  IN NO EVENT SHALL THE COPYRIGHT OWNER OR
+# CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL,
+# EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO,
+# PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR
+# PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY
+# OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+# (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+# OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+
+from typing import Any, Dict
+
+from genai_perf.export_data.exporter_config import ExporterConfig
+from genai_perf.logging import logging
+from prometheus_client import Gauge, start_http_server
+
+logger = logging.getLogger(__name__)
+
+
+class PrometheusExporter:
+    """
+    Exports metrics in a Prometheus-compatible format via an HTTP endpoint.
+    """
+
+    def __init__(self, config: ExporterConfig):
+        self._stats = config.stats
+        self._metrics = config.metrics
+        self._args = config.args
+        if not hasattr(self._args, "prometheus_port"):
+            raise ValueError("Prometheus port is not provided.")
```

Comment:
This check is not necessary, as the lines below (`47-51`) will assign `self._port` to the default value if it is not found in `self._args`. The way it is written now, the else condition on lines `49-50` will never be executed because a `ValueError` will be raised.

Reply:
Great point, will fix this. Thanks Anthony!